File: 2021/2021W43/PreppindataW43.py
import pandas as pd
import datetime

xlsx = pd.ExcelFile(r"C:\Data\PreppinData\2021W43 Input.xlsx")

unit_a = pd.read_excel(xlsx, 'Business Unit A ')
unit_a = unit_a.merge(pd.read_excel(xlsx, 'Risk Level'), left_on='Rating', right_on='Risk level')
unit_a['Date lodged'] = unit_a.apply(lambda x: f"{x['Date']}/{x['Month ']}/{x['Year']}", axis=1)
unit_a = unit_a[['Date lodged', 'Status', 'Risk rating']].rename(columns={'Risk rating':'Rating'})
unit_b = pd.read_excel(xlsx, 'Business Unit B ', skiprows=5)[['Date lodged', 'Status', 'Rating']]

rating = pd.concat([unit_a, unit_b]).rename(columns={'Status':'Status by Process'})
rating['Status by Lodged'] = rating['Date lodged'].apply(lambda x: 'Opening cases' 
                                                         if pd.to_datetime(x, format='%d/%m/%Y').date()<datetime.date(2021,10,1) else 'New cases')
rating = rating.melt(id_vars=['Rating','Date lodged'], value_name='Status', var_name='ToDrop').drop('ToDrop', axis=1)
rating = rating.pivot_table(index='Rating', columns='Status', values='Date lodged', aggfunc='count').reset_index().fillna(0)
rating = rating.rename(columns={'In Progress':'Continuing'})
rating = rating.melt(id_vars='Rating', value_name='Cases', var_name='Status').sort_values(['Rating','Status']) 

File: 2021/2021W43/preppindataW43.R
library(readxl)
library(dplyr)
library(tidyr)

input <- "C:/Data/PreppinData/2021W43 Input.xlsx"

final <- read_excel(input,'Business Unit A ') %>%
  mutate('Date lodged' = paste(`Date`,`Month`,`Year`, sep='/')) %>%
  merge(., read_excel(input,'Risk Level'), by.x='Rating', by.y='Risk level') %>%
  select(c('Risk rating', 'Date lodged', 'Status')) %>%
  rename('Rating' = 'Risk rating') %>%
  rbind(., read_excel(input,'Business Unit B ', skip = 5) %>%
          select(c('Rating', 'Date lodged', 'Status'))) %>%
  rename('Status by Process' = 'Status') %>%
  mutate('Status by Lodged' = if_else(as.Date(`Date lodged`,'%d/%m/%Y')<as.Date('1/10/2021','%d/%m/%Y'), 'Opening cases','New cases')) %>%
  pivot_longer(cols=starts_with('Status'), names_to='Type', values_to = 'Status') %>%
  pivot_wider(id_cols='Rating', names_from='Status', values_from='Date lodged', values_fn = length) %>%
  rename('Continuing' = 'In Progress') %>%
  pivot_longer(cols=-'Rating', names_to='Status', values_to = 'Cases') %>%
  mutate_at(vars('Cases'), ~replace_na(., 0)) %>%
  arrange(`Rating`, `Status`)

View(final)
